tests/unit/test_sizes.py

```diff
@@ -120,54 +120,56 @@ def test_grid_parameters_from_harmony_message_has_complete_information(self):
         self.assertDictEqual(expected_parameters, actual_parameters)
 
     def test_grid_parameters_from_harmony_no_message_information(self):
-        """input granule is in preferred_crs on a 25km grid
-
-        But has grid extents different from the ICD defaults.
-
-        """
+        """input granule is in preferred_crs on a 25km grid"""
+        crs = CRS.from_epsg(sp_seaice_grid['epsg'])
+        height = sp_seaice_grid['height']
+        width = sp_seaice_grid['width']
+        img_transform = Affine.translation(
+            sp_seaice_grid['left'], sp_seaice_grid['top']
+        ) * Affine.scale(sp_seaice_grid['xres'], -1 * sp_seaice_grid['yres'])
 
-        preferred_scale_extent = {
-            'xmin': -4194304.0,
-            'ymin': -4194304.0,
-            'xmax': 4194304.0,
-            'ymax': 4194304.0,
+        left, bottom, right, top = rasterio.transform.array_bounds(
+            height, width, img_transform
+        )
```

Comment:
So first, this is using `array_bounds()`, not `transform_bounds()`.  So it's not the exact same code tested.  Second the `array_bounds()` function is just doing Affine transformation math. The code for computing the bounds found in Rasterio's codebase is  

``` python
    a, b, c, d, e, f, _, _, _ = transform
    if b == d == 0:
        west, south, east, north = c, f + e * height, c + a * width, f
```

I could change it to that, but would that make it any clearer?  I don't think so.  I do understand the concern, but I do think if I put in the math there, it doesn't actually clarify it any.  If I had memorized that computation I might have used it instead of the function, but I didn't have it memorized.

Reply:
Sorry for missing the different function call there, good point.

Yeah, I think this is good. It just felt like a little bit of a code-smell type thing. Probably would have not caught me quite as much if I'd spotted it was a different function.